### DISCUSSION TITLE
OCI sync from inside the cluster

### DISCUSSION THREAD

**Discussion by mo-saeed** [fluxcd/flux2#4873] [General] [answered]
Hi, 

I am trying to bootstrap flux in EKS cluster using terraform, everything works fine as I have network connection from my local machine to the gitlab server. The problem is syncing the repo from inside the cluster, it fails as theres not connection from inside the cluster to my gitlab server.

is there any clean workaround to change the sync configuration to use OCI without the need to wait for this RFC https://github.com/fluxcd/flux2/pull/4749
Thanks a lot in advance !

**Comment by stefanprodan** [comment]
The Flux Operator has this capability to sync the cluster from OCI, see https://fluxcd.control-plane.io/operator/flux-sync/#sync-from-a-container-registry

**Reply by mo-saeed** [reply]
yes that i know. but how to configure that during the bootstrap for the bootstrap project? i get this file gotk-sync.yml which is not supposed to be edited. 

What happens if I edited this file ? would it work ?

**Reply by stefanprodan** [reply, marked as answer]
Bootstrap does not support OCI sources, so you either use the Flux Operator, write your own bootstrapping tool or wait for the RFC to be implemented.

**Reply by mo-saeed** [reply]
is the flux-operator the preferred and recommended way to install Flux ? Thats a feeling I get from the docs reading.

**Reply by stefanprodan** [reply]
It is the recommended way when you buy Flux Enterprise from ControlPlane. For CNCF Flux you have lots of options, including writing your own installer.

**Reply by mo-saeed** [reply]
I see flux-operator is opensource, do you think is not a good idea to use incase of no intention to buy Flux Enterprise ?

**Reply by mo-saeed** [reply]
> Bootstrap does not support OCI sources, so you either use the Flux Operator, write your own bootstrapping tool or wait for the RFC to be implemented.

Do you have an estimation when the RFC will be implementer and released ?

**Reply by stefanprodan** [reply]
> I see flux-operator is opensource, do you think is not a good idea to use incase of no intention to buy Flux Enterprise ?

You can use it, that's why we made it open source.

> Do you have an estimation when the RFC will be implementer and released ?

No I don't. It's not up to me, the RFC must be reviewed and approved by other Flux maintainers. After the RFC is merged, we'll need for someone to dedicate lots of time and work on the implementation.
